packages/api/src/controllers/asset.ts

```diff
@@ -267,8 +267,16 @@ function getDownloadUrl(
   { vodObjectStoreId }: Request["config"],
   ingest: string,
   asset: WithID<Asset>,
-  os: ObjectStore
+  os: ObjectStore,
+  useStaticRenditions: boolean
 ): string {
+  if (useStaticRenditions) {
```

Comment:
Sorry I know I suggested this myself, but I think we need to make this conditional on the `asset` object only, otherwise when this function is executed through the asset API flow, it will not return the correct download URL for the recording-based asset.

So I think we need sth like:
```suggestion
  if (asset.source?.type === "recording" && !asset.playbackRecodingId) {
```
This will match only the assets that come from the new recordings v2 flow (which went through the `import` flow and always had that `playbackRecordingId` field set). Then we also don't need the boolean arg anymore. WDYT?

Reply:
Updated in 290b6c98cd9cc1edfd2ec7686f137b7119199bd9